### ISSUE TITLE
wrong answer with infeasible mipstart

### ISSUE THREAD

**Issue by zengberg** [coin-or/Cbc#449]
Hello,

I'm seeing strange behavior when using CBC 2.10.5 (Debian Bullseye) with and without passing in `mipstart` data. The issue is not present on CBC 2.9.9 (Debian Buster).

Running
```
cbc -import problem.lp -mipstart mipstart.soln -solve
```
gives me an incorrect solution to the problem at hand, whereas running without `mipstart`
```
cbc -import problem.lp -solve
```
gives me the correct solution. I've attached both `problem.lp` and `mipstart.soln`.

For this particular problem, the "solution" in `mipstart.soln` is infeasible. Moreover, the CBC logs indicate that CBC is aware of this:

```
> Cbc0045I Warning: mipstart values could not be used to build a solution.
```

Full logs pasted below.

```
$ cbc -import problem.lp -mipstart mipstart.soln -solve
Welcome to the CBC MILP Solver 
Version: 2.10.5 
Build Date: Nov 17 2020 

command line - cbc -import problem.lp -mipstart mipstart.soln -solve (default strategy 1)
opening mipstart file ./mipstart.soln.
MIPStart values read for 3 variables.
Continuous objective value is -10.4581 - 0.00 seconds
Cgl0003I 60 fixed, 0 tightened bounds, 0 strengthened rows, 0 substitutions
Cgl0005I 3 SOS with 39 members
Cgl0004I processed model has 6 rows, 40 columns (39 integer (39 of which binary)) and 102 elements
Cbc0045I Trying just fixing integer variables (and fixingish SOS).
Cbc0045I Warning: mipstart values could not be used to build a solution.
Cbc0038I Initial state - 1 integers unsatisfied sum - 0.420084
Cbc0038I Solution found of -9.05774
Cbc0038I Branch and bound needed to clear up 1 general integers
Cbc0038I Full problem 6 rows 40 columns, reduced to 4 rows 24 columns
Cbc0038I Cleaned solution of -6.03847
Cbc0038I Before mini branch and bound, 12 integers at bound fixed and 3 continuous
Cbc0038I Full problem 6 rows 40 columns, reduced to 2 rows 12 columns
Cbc0038I Mini branch and bound did not improve solution (0.01 seconds)
Cbc0038I Round again with cutoff of -6.34041
Cbc0038I Reduced cost fixing fixed 24 variables on major pass 2
Cbc0038I Solution found of -9.05774
Cbc0038I Branch and bound needed to clear up 1 general integers
Cbc0038I Mini branch and bound could not fix general integers
Cbc0038I No solution found this major pass
Cbc0038I Before mini branch and bound, 12 integers at bound fixed and 3 continuous
Cbc0038I Mini branch and bound did not improve solution (0.01 seconds)
Cbc0038I After 0.01 seconds - Feasibility pump exiting with objective of -6.03847 - took 0.00 seconds
Cbc0012I Integer solution of -6.0384741 found by feasibility pump after 0 iterations and 0 nodes (0.01 seconds)
Cbc0006I The LP relaxation is infeasible or too expensive
Cbc0013I At root node, 0 cuts changed objective from -9.0577383 to -9.0577383 in 1 passes
Cbc0014I Cut generator 0 (Probing) - 1 row cuts average 0.0 elements, 6 column cuts (6 active)  in 0.000 seconds - new frequency is 1
Cbc0014I Cut generator 1 (Gomory) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 2 (Knapsack) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 3 (Clique) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 4 (MixedIntegerRounding2) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 5 (FlowCover) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 6 (TwoMirCuts) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0014I Cut generator 7 (ZeroHalf) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.000 seconds - new frequency is -100
Cbc0001I Search completed - best objective -6.038474112617573, took 0 iterations and 0 nodes (0.01 seconds)
Cbc0035I Maximum depth 0, 7 variables fixed on reduced cost
Cuts at root node changed objective from -9.05774 to -9.05774
Probing was tried 1 times and created 7 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Gomory was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Knapsack was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Clique was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
MixedIntegerRounding2 was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
FlowCover was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
TwoMirCuts was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
ZeroHalf was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)

Result - Optimal solution found

Objective value:                -6.03847411
Enumerated nodes:               0
Total iterations:               0
Time (CPU seconds):             0.01
Time (Wallclock seconds):       0.01

Total time (CPU seconds):       0.01   (Wallclock seconds):       0.01
```

```
$ cbc -import problem.lp -solve                        

Welcome to the CBC MILP Solver 
Version: 2.10.5 
Build Date: Nov 17 2020 

command line - cbc -import problem.lp -solve (default strategy 1)
Continuous objective value is -10.4581 - 0.00 seconds
Cgl0003I 60 fixed, 0 tightened bounds, 0 strengthened rows, 0 substitutions
Cgl0005I 3 SOS with 39 members
Cgl0004I processed model has 6 rows, 40 columns (39 integer (39 of which binary)) and 102 elements
Cbc0038I Initial state - 0 integers unsatisfied sum - 2.22045e-16
Cbc0038I Solution found of -9.05774
Cbc0038I Relaxing continuous gives -9.05774
Cbc0038I Before mini branch and bound, 36 integers at bound fixed and 3 continuous
Cbc0038I Mini branch and bound did not improve solution (0.00 seconds)
Cbc0038I After 0.00 seconds - Feasibility pump exiting with objective of -9.05774 - took 0.00 seconds
Cbc0012I Integer solution of -9.0577383 found by feasibility pump after 0 iterations and 0 nodes (0.00 seconds)
Cbc0001I Search completed - best objective -9.057738264750917, took 0 iterations and 0 nodes (0.00 seconds)
Cbc0035I Maximum depth 0, 0 variables fixed on reduced cost
Cuts at root node changed objective from -9.05774 to -9.05774
Probing was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Gomory was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Knapsack was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Clique was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
MixedIntegerRounding2 was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
FlowCover was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
TwoMirCuts was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
ZeroHalf was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
1 bounds tightened after postprocessing


Result - Optimal solution found

Objective value:                -9.05773826
Enumerated nodes:               0
Total iterations:               0
Time (CPU seconds):             0.01
Time (Wallclock seconds):       0.01

Total time (CPU seconds):       0.01   (Wallclock seconds):       0.02
```
[problem.lp.txt](https://github.com/coin-or/Cbc/files/7400128/problem.lp.txt)
[mipstart.soln.txt](https://github.com/coin-or/Cbc/files/7400126/mipstart.soln.txt)

**Comment by jjhforrest** [comment]
Thanks,

Can reproduce bug - will investigate.

John Forrest

**Comment by jjhforrest** [comment]
Still looking into it.

I can fix but want to do it in best way.

Problem is to do with Cbc finding SOS during preprocessing and getting 
too excited.

While I think, I suggest you use

-preprocess on  (default is sos)

John Forrest

**Comment by zengberg** [comment]
Thank you John -- with `-preprocess on`, Cbc does the right thing.

**Comment by zengberg** [comment]
This is fixed on CBC 2.10.7.
